File: BackendDotNet/Models/BloodInventory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackendDotNet.Models
{
    [Table("blood_inventory")]
    public class BloodInventory
    {
        [Key]
        [Column("inventory_id")]
        public int InventoryId { get; set; }

        [Required]
        [Column("blood_group")]
        [StringLength(5)]
        public string BloodGroup { get; set; } = string.Empty;

        [Required]
        [Column("units")]
        public int Units { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [Column("status_id")]
        public int StatusId { get; set; } = 1;

        // Constructor for service usage
        public BloodInventory() { }

        public BloodInventory(string bloodGroup)
        {
            BloodGroup = bloodGroup;
        }
    }
}

File: backend spring/src/main/java/com/bloodbank/entity/BloodInventory.java
package com.bloodbank.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "blood_inventory")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BloodInventory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inventory_id")
    private Integer inventoryId;

    @Column(name = "blood_group", nullable = false, unique = true, length = 5)
    private String bloodGroup;

    @Column(nullable = false)
    private int units;

    // Extra Columns from our DB design
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @Column(name = "status_id", nullable = false)
    private int statusId = 1; // e.g., 1-OK, 2-Low, etc.

    // Custom constructor needed by our service
    public BloodInventory(String bloodGroup) {
        this.bloodGroup = bloodGroup;
    }
}